mlir/test/Conversion/VectorToSPIRV/vector-to-spirv.mlir

```diff
@@ -274,7 +274,7 @@ func.func @insert_dynamic(%val: f32, %arg0 : vector<4xf32>, %id : index) -> vect
 //       CHECK:   spirv.CompositeInsert %[[VAL]], %[[V]][2 : i32] : f32 into vector<4xf32>
 func.func @insert_dynamic_cst(%val: f32, %arg0 : vector<4xf32>) -> vector<4xf32> {
   %idx = arith.constant 2 : index
-  %0 = vector.insert %val, %arg0[%idx] : f32 into vector<4xf32>
+  %0 = vector.insert %val, %arg0[%idx : index] : f32 into vector<4xf32>
```

Comment:
Could you also add tests for i8 and maybe i1 as the insert/extract index types? These require type conversion in the general case.

Reply:
Hey Jakub, I added i32 and i8 tests but they need to modify the spirv conversion, as you mentioned. Would you mind helping with that? I have no idea about that pass. How would you like to proceed?